### ISSUE TITLE
JavaFX QuantumRenderer bug on Linux when running Jar

### ISSUE THREAD

**Issue by nichee** [nus-cs2103-AY2324S2/forum#448]
Thanks for opening my thread :)
My Launcher.java file works fine on windows and mac, but crashes on Linux with the QuantumRenderer error.

It's especially odd as I've tested the Launcher.jar on windows and a macbook pro and it works on both, but it seems throw the exact same QuantumRenderer error on a normal macbook. Edit: The macbook didn't have the Zulu 11 with javaFX, its fixed now!

I suspect it has something to do with my JDK and/or gradle JVM settings, I'm currently using Amazon corretto 11.0.22, and I've verified the gradle JVM is the same as the Intellij JDK. I've pasted the build.gradle(sorry about the weird GFMD formatting)

edit: I've tried like 5 different JDKs including ones bundled with javaFX already, the same error still exists

![Screenshot_5](https://github.com/nus-cs2103-AY2324S2/forum/assets/45154216/44c2aee6-2dfa-4e39-ae56-73d4571b4b4f)
![Screenshot_6](https://github.com/nus-cs2103-AY2324S2/forum/assets/45154216/f2af3033-25a9-4b96-891b-9837cc91e980)

build.gradle
```
plugins {
    id 'java'
    id 'application'
    id 'com.github.johnrengelman.shadow' version '7.1.2'
    id 'org.openjfx.javafxplugin' version '0.0.8'
}

repositories {
    mavenCentral()
}
javafx {
    version = "17.0.7"
    modules = [ 'javafx.controls' ]
}
sourceCompatibility = 11
targetCompatibility = 11


dependencies {
    String javaFxVersion = '17.0.7'

    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'linux'

    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.10.0'
    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.10.0'
}


test {
    useJUnitPlatform()
    testLogging {
        events "passed", "skipped", "failed"

        showExceptions true
        exceptionFormat "full"
        showCauses true
        showStackTraces true
        showStandardStreams = false
    }
}

application {
    mainClass.set('shirmin.Launcher')
}

shadowJar {
    archiveBaseName = "Launcher"
    archiveClassifier = null
    dependsOn("distZip", "distTar")
    mainClassName = 'shirmin.Launcher'
    archiveFileName = 'Launcher.jar'
}

run{
    standardInput = System.in
}
```

**Comment by damithc** [comment]
@nichee While waiting for inputs from others, you may want to use [fenced code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) to format your post for better readability.

**Comment by nichee** [comment]
Fixed! Genuinely not sure what I changed that made it work - I was testing on a macbook without the right JDK so I changed 50 different things and installed a new JDK 5 times. But basically 8 hours of changes were pointless because all my Jars from the past 8 hours work on linux now. I love coding :)

added my new build.gradle in case it helps someone

```
plugins {
    id 'java'
    id 'application'
    id 'com.github.johnrengelman.shadow' version '7.1.2'
}

sourceCompatibility = 11
targetCompatibility = 11

repositories {
    mavenCentral()
}



dependencies {
    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.10.0'
    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.10.0'

    String javaFxVersion = '17.0.7'

    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'linux'

}


test {
    useJUnitPlatform()

    testLogging {
        events "passed", "skipped", "failed"

        showExceptions true
        exceptionFormat "full"
        showCauses true
        showStackTraces true
        showStandardStreams = false
    }
}

application {
    mainClass.set('shirmin.Launcher')
}

shadowJar {
    archiveBaseName = "Launcher"
    archiveClassifier = null
    dependsOn("distZip", "distTar")
    mainClassName = 'shirmin.Launcher'
    archiveFileName = 'Launcher.jar'
}

run{
    standardInput = System.in
}
```

**Comment by damithc** [comment]
Glad to hear you managed to fix it on your own @nichee 
So, in the Linux machine you used for testing, the old JAR produces the same error but the new JAR doesn't?

**Comment by nichee** [comment]
Yep! Tested the old Jar and it still produces the same error. Its really really strange because I'm 95% sure it wasn't a jdk/jvm issue. 
My best guess at this stage is there may have been some problem with the JavaFX version - I was previously using 17.0.8, but changed it to 17.0.7.

Still idk tho, why would it work in every OS through both IDE and JAR except linux. all the other quantumrenderer bugs on this forum was problems with the terminal/running in ide/wrong jdk versions so it was really hard to debug.
oh well.

edit: forgot to mention i used Ubuntu on oracle VM virtualbox, which produced the exact same error message as the automated smoke test in the image above. 

edit 2: oh and forgot to add I switched to Azul zulu openjdk 11.  May have been some strange interaction with amazon corretto that caused the linux bug. idk.
